src/types/Hashed.ts

```diff
@@ -0,0 +1,29 @@
+import * as bcryptC from 'bcrypt'
+import * as Promise from 'bluebird'
+import * as _ from 'lodash'
+
+const bcrypt:any = Promise.promisifyAll(bcryptC)
```

Comment:
This is ugly but it works as an easy fix. 
https://github.com/Microsoft/TypeScript/issues/8685 also contains a better solution but it would be nice have a single source of truth, importable from different modules which extends `bcrypt` or any commonly promisified library with the `*Async` signatures.